src/CoreShop/Component/Order/Model/CartInterface.php

```diff
@@ -54,19 +54,35 @@ public function hasPriceRules();
     /**
      * @param $priceRule
      */
-    public function addPriceRule($priceRule);
+    public function addPriceRule(ProposalCartPriceRuleItemInterface $priceRule);
```

Comment:
this changes the signature to actually fit the expected type -> at least the new expected type of what we store in the Fieldcollection